api/v1alpha4/virtualmachineclassinstance_types.go

```diff
@@ -0,0 +1,51 @@
+// © Broadcom. All Rights Reserved.
+// The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha4
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VirtualMachineClassInstanceSpec defines the desired state of VirtualMachineClassInstance.
+// It is a composite of VirtualMachineClassSpec.
+type VirtualMachineClassInstanceSpec struct {
+	VirtualMachineClassSpec `json:",inline"`
```

Comment:
We would need something in the instance spec to indicate the hash.  During each VM class reconcile:
1. The controller would need to calculate the hash from the spec of the class
2. Get all existing instances for the class (using ownerrefs)
3. If an instance for this hash exists, mark it as active by applying the label.  If it is already active, then no-op.
4. If any other instance has the active label, that is now removed.

For step 3, we can get the class spec from the instance spec and re-calculate the hash.  But it will be massively more performant to just have the configspec hash in the spec -- specially since it is never supposed to change.

Another approach would be to rely on the naming (vm-class-display-name + SHA), but that seems brittle.